src/components/App.tsx

```diff
@@ -2365,7 +2397,27 @@ class App extends React.Component<AppProps, AppState> {
     state,
   ) => {
     this.cancelInProgresAnimation?.();
-    this.setState(state);
+
+    // TODO follow-participant
+    // seems to work fine but ideally figure out a way to satisfy TS
+    // @ts-ignore
+    this.setState((prevState, props) => {
+      if (typeof state === "function") {
+        return {
+          ...state(prevState, props),
+          userToFollow: prevState.shouldDisconnectFollowModeOnCanvasInteraction
+            ? null
+            : prevState.userToFollow,
+        };
+      }
+
+      return {
+        ...state,
+        userToFollow: prevState.shouldDisconnectFollowModeOnCanvasInteraction
+          ? null
+          : prevState.userToFollow,
+      };
+    });
```

Comment:
```suggestion
    // potentially unfollow participant
    this.setState((prevState, props) => {
      return {
        ...(typeof state === "function" ? state(prevState, props) : state),
        userToFollow: prevState.shouldDisconnectFollowModeOnCanvasInteraction
          ? null
          : prevState.userToFollow,
      };
    });
```

works for me without the ts-ignore.

Also, I'd simplify the code as per above.